configure.ac

```diff
@@ -383,61 +383,167 @@ m4_ifdef([PKG_PROG_PKG_CONFIG], [
+   curses_pkg_names=""
+   htop_check_libcurses_capability "$CURSES_CFLAGS" "$CURSES_LIBS"
+fi
+
+m4_ifdef([PKG_PROG_PKG_CONFIG], [
+# Prioritize $PKG_CONFIG over ncurses*-config, as users might need to
+# cross-compile htop.
+if test "x$PKG_CONFIG" != x; then
+   for curses_name in $curses_pkg_names; do
+      echo retrieving $curses_name configuration through $PKG_CONFIG >&AS_MESSAGE_LOG_FD
+      $PKG_CONFIG --exists --print-errors $curses_name >&AS_MESSAGE_LOG_FD 2>&AS_MESSAGE_LOG_FD || continue
+
+      : # Resets "$?" to 0
+      htop_config_cflags=`$PKG_CONFIG --cflags $curses_name 2>/dev/null` || continue
+      : # Resets "$?" to 0
+      htop_config_libs=`$PKG_CONFIG --libs $curses_name 2>/dev/null` || continue
+
+      AC_MSG_RESULT([found $curses_name configuration through $PKG_CONFIG])
+
+      if htop_check_libcurses_capability "$htop_config_cflags" "$htop_config_libs"; then
+         break
+      fi
+   done
+fi
 ])
 
-AC_ARG_ENABLE([unicode],
-              [AS_HELP_STRING([--enable-unicode],
-                              [enable Unicode support @<:@default=yes@:>@])],
-              [],
-              [enable_unicode=yes])
-if test "x$enable_unicode" = xyes; then
-   HTOP_CHECK_SCRIPT([ncursesw6], [waddwstr], [HAVE_LIBNCURSESW], "ncursesw6-config",
-    HTOP_CHECK_SCRIPT([ncursesw], [waddwstr], [HAVE_LIBNCURSESW], "ncursesw6-config",
-     HTOP_CHECK_SCRIPT([ncursesw], [wadd_wch], [HAVE_LIBNCURSESW], "ncursesw5-config",
-      HTOP_CHECK_SCRIPT([ncurses], [wadd_wch], [HAVE_LIBNCURSESW], "ncurses5-config",
-       HTOP_CHECK_LIB([ncursesw6], [addnwstr], [HAVE_LIBNCURSESW],
-        HTOP_CHECK_LIB([ncursesw], [addnwstr], [HAVE_LIBNCURSESW],
-         HTOP_CHECK_LIB([ncurses], [addnwstr], [HAVE_LIBNCURSESW],
-          AC_MSG_ERROR([can not find required library libncursesw; you may want to use --disable-unicode])
-   )))))))
+if test "$htop_curses_pkg_found" = no; then
+   for curses_name in $curses_pkg_names; do
+      echo retrieving $curses_name configuration through ${curses_name}-config >&AS_MESSAGE_LOG_FD
 
+      ${curses_name}-config --cflags >/dev/null 2>&AS_MESSAGE_LOG_FD || continue
+
+      : # Resets "$?" to 0
+      htop_config_cflags=`${curses_name}-config --cflags 2>/dev/null` || continue
+      : # Resets "$?" to 0
+      htop_config_libs=`${curses_name}-config --libs 2>/dev/null` || continue
+
+      AC_MSG_RESULT([found $curses_name configuration through ${curses_name}-config])
+
+      if htop_check_libcurses_capability "$htop_config_cflags" "$htop_config_libs"; then
+         break
+      fi
+   done
+fi
+
+if test "$htop_curses_pkg_found" = no; then
+   for curses_name in $curses_pkg_names; do
+      echo trying simple linking with "-l$curses_name" >&AS_MESSAGE_LOG_FD
+      if htop_check_libcurses_capability "" "-l$curses_name"; then
+         break
+      fi
+   done
+fi
+
+if test "$htop_curses_pkg_found" = no; then
+   AC_MSG_ERROR([cannot link to curses library])
```

Comment:
In the current `main` of `configure`, it might suggest users to specify `--disable-unicode` in the error message. My PR didn't yet have this usability feature. Should I add it?

Reply:
How much work is it to re-run the detection code with `--disable-unicode` if not provided explicitly in either direction? But as a first step, providing this hint might suffice.

Reply:
@BenBE Not sure what you mean by "direction".

Since htop documentation already states that `--enable-unicode` would be the default if unspecified (it's `yes`, not `check`). I cannot make automatic fallback to `--disable-unicode` but to tell the user to specify such. Since `htop_check_libcurses_capability` checks _libncurses_ features first before _libncursesw_ features. Adding the `--disable-unicode` suggestion message is as easy as adding a shell variables noting that _libncurses_ is found but not _libncursesw_.

Reply:
Yes, I was refering to `check` vs `yes`/`no`.

Also, shouldn't it be `libncursesw` before `libncurses` (based on the old code)?

Reply:
@BenBE The order swapping in `htop_check_libcurses_capability` is necessary to tell that, "oh, I found a working ncurses library, but this not doesn't support wide character functions".

Reply:
Ah, k.